### PR TITLE
Add example of CoreDNS extension

### PR DESCRIPTION
**Pull request by pipo02mix**
Towards https://github.com/giantswarm/giantswarm/issues/23856

### CODE REVIEW

**Review by pipo02mix — COMMENTED** [review]
noice

**Comment by pipo02mix** [comment]
@uvegla is good to go?

**Comment by pipo02mix** [comment]
I dont think this change can provoke kind test to fail, wdyt?

File: management-clusters/MC_NAME/organizations/ORG_NAME/workload-clusters/WC_NAME/mapi/cluster/default_apps_user_config.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+                  log
+                  errors
+                  cache 30
+                  loop
+                  reload
+                  loadbalance
+                }
+kind: ConfigMap
+metadata:
+  name: ${cluster_name}-default-apps-userconfig
```
Comment (piontec) [inline management-clusters/MC_NAME/organizations/ORG_NAME/workload-clusters/WC_NAME/mapi/cluster/default_apps_user_config.yaml:20]: I think the problem might be that the `namespace` here is missing

**Review by piontec — COMMENTED** [review]
I think tests fail for a reason, something can't be successfully deployed. Check the missing `namespace` on ConfigMap's meta, I think that's it

File: docs/add_wc_structure.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ An example of a WC cluster directory structure is available in [WC_NAME/](/manag
 ## Export environment variables
 
 **Note**, Management Cluster codename, Organization name and Workload Cluster name are needed in multiple places across
-this instruction, the least error prone way of providing them is by exporting as environment variables:
+this instruction, the least error-prone way of providing them is by exporting as environment variables:
```
Comment (mproffitt) [inline docs/add_wc_structure.md:29]: Grammatically I find this hard to read and think it sounds a little clumsy.

> In order to ensure consistency in the execution of this instruction, consider exporting the following environment variables for easy reference. These represent common names used throughout these commands such as Management Cluster codename, Organisation name and Workload cluster name.
> 
> ```bash
> export MC_NAME="CODENAME"
> export ORG_NAME="ORGANIZATION"
> export WC_NAME="CLUSTER_NAME"
> export GIT_REPOSITORY_NAME="REPOSITORY_NAME"
> ```

File: docs/add_wc_structure.md [hunk 3]
```diff
@@ -337,6 +337,64 @@ Kubernetes Secret, you MUST not create multiple Secrets.
...
+                      log
+                      errors
+                      cache 30
+                      loop
+                      reload
+                      loadbalance
+                  }
+   kind: ConfigMap
+   metadata:
+   name: ${cluster_name}-default-apps-userconfig
```
Comment (pipo02mix) [inline docs/add_wc_structure.md:370]:
```suggestion
   name: ${cluster_name}-default-apps-userconfig
   namespace: org-${organization}
```

File: management-clusters/MC_NAME/organizations/ORG_NAME/workload-clusters/WC_NAME/mapi/cluster/default_apps_user_config.yaml (hunk 1 above)
Comment (pipo02mix) [inline management-clusters/MC_NAME/organizations/ORG_NAME/workload-clusters/WC_NAME/mapi/cluster/default_apps_user_config.yaml:20]:
```suggestion
  name: ${cluster_name}-default-apps-userconfig
  namespace: org-${organization}
```

File: docs/add_wc_structure.md (hunk 2 above)
Comment (pipo02mix) [reply docs/add_wc_structure.md:29]: But from where did you take that text? I don't see that in my PR 🤔

File: management-clusters/MC_NAME/organizations/ORG_NAME/workload-clusters/WC_NAME/mapi/cluster/default_apps_user_config.yaml (hunk 1 above)
Comment (pipo02mix) [reply management-clusters/MC_NAME/organizations/ORG_NAME/workload-clusters/WC_NAME/mapi/cluster/default_apps_user_config.yaml:20]: in theory there is a default namespace equal to org namespace by I have added just in case

File: docs/add_wc_structure.md (hunk 2 above)
Comment (mproffitt) [reply docs/add_wc_structure.md:29]: What I put in quote is my rephrasing of the whole **note** section

Comment (mproffitt) [reply docs/add_wc_structure.md:29]: IMO I would rephrase  the whole of the following to what I offered in the earlier comment

```
**Note**, Management Cluster codename, Organization name and Workload Cluster name are needed in multiple places across
this instruction, the least error prone way of providing them is by exporting as environment variables:
```

Comment (pipo02mix) [reply docs/add_wc_structure.md:29]: ahh ok thx
you can use 
Screenshot 2022-10-04 at 13 19 43
 this button to offer suggestions

**Comment by pipo02mix** [comment]
@piontec it worked now!

**Review by uvegla — APPROVED** [review]
